metric/registry.go

```diff
@@ -67,15 +68,17 @@ func (r *Registry) initGauge(g *gauge, labelKeys []string, name string, descript
 		Unit:        unit,
 		LabelKeys:   labelKeys,
 	}
-	r.gauges[name] = g
+	r.gauges.Store(name, g)
 	return g
 }
 
-// ReadAll reads all gauges in this registry and returns their values as metrics.
-func (r *Registry) ReadAll() []*metricdata.Metric {
-	ms := make([]*metricdata.Metric, 0, len(r.gauges))
-	for _, g := range r.gauges {
+// Read reads all gauges in this registry and returns their values as metrics.
+func (r *Registry) Read() []*metricdata.Metric {
```

Comment:
Why change this name? I don't feel strongly about it but was curios.

Reply:
Producer [interface](https://github.com/census-instrumentation/opencensus-go/blob/2eaaf3ad5aeaf6d5f513040ace947fad79d9da63/metric/metricproducer/producer.go#L27) is Read(). Since there is only going to be one Read there is no point in naming ReadAll. (short is better as long as it is clear). 
If ReadAll is preferable I can change the producer interface.